api/handlers.go

```diff
@@ -1143,6 +1161,297 @@ func SearchStatsHandler(c *gin.Context) {
+	}
+
+	// The logic up to this point is the same as the regular search handle
+
+	se := search.NewESEngine(esc, db, cacheM, tc, variables, consts.ES_MOBILE_SEARCH_RESULT_TYPES)
+
+	checkTypo := false // Currently not supported in mobile
+	searchTweets := c.Query("search_tweets") == "true"
+	searchLessonSeries := c.Query("search_lesson_series") == "true"
+
+	res, err := se.DoSearch(
+		context.TODO(),
+		query,
+		sortByVal,
+		from,
+		size,
+		preference,
+		checkTypo,
+		searchTweets,
+		searchLessonSeries,
+		false, // Highlights are not currently supported in mobile
+		time.Duration(0),
+	)
+
+	if err == nil {
+		// TODO: How does this slows the search query? Consider logging in parallel.
+		if !query.Deb {
+			err = logger.LogSearch(query, sortByVal, from, size, searchId, suggestion, res, se.ExecutionTimeLog)
+			if err != nil {
+				log.Warnf("Error logging search: %+v %+v", err, res)
+			}
+		}
+
+		mapIdsByType := map[string][]string{}
+		mobileRespItemMap := map[string]*MobileSearchResponseItem{}
+		allItems := []*MobileSearchResponseItem{}
+
+		imagesUrlTemplate := viper.GetString("content_unit_images.url_template")
+
+		for _, hit := range res.SearchResult.Hits.Hits {
+
+			var result es.Result
+			if hit.Source == nil {
+				search.LogIfDeb(&query, fmt.Sprintf("Empty source in hit: %+v.", hit))
+				continue
+			}
+			json.Unmarshal(*hit.Source, &result)
+			if err != nil {
+				NewInternalError(err).Abort(c)
+				return
+			}
+			var mobileResp *MobileSearchResponseItem
+			var date *time.Time = nil
+
+			if result.EffectiveDate != nil {
+				date = &result.EffectiveDate.Time
+			}
+			if hit.Type == "result" {
+				switch result.ResultType {
+				case consts.ES_RESULT_TYPE_UNITS:
+					var image *string
+					var ct string
+					isArticle := cacheM.SearchStats().IsContentUnitTypeArticle(result.MDB_UID)
+					if isArticle {
+						ct = consts.CT_ARTICLES
+					} else {
+						ct = result.ResultType
+						imageStr := fmt.Sprintf(imagesUrlTemplate, result.MDB_UID)
+						image = &imageStr
+					}
+					mobileResp = &MobileSearchResponseItem{
+						ContentUnitUid: &result.MDB_UID,
+						Image:          image,
+						Title:          result.Title,
+						Date:           date,
+						Type:           ct,
+					}
+
+				case consts.ES_RESULT_TYPE_COLLECTIONS:
+					var image *string
+					firstUnit := cacheM.SearchStats().GetCollectionFirstUnit(result.MDB_UID)
+					isArticle := firstUnit != nil && cacheM.SearchStats().IsContentUnitTypeArticle(*firstUnit)
+					if isArticle {
+						// Articles collection is not currently supported in mobile
+						search.LogIfDeb(&query, "Skip result for mobile search: Articles Collection.")
+						continue
+					}
+					if firstUnit != nil {
+						// We retrieve collection image according to the first content unit of the collection.
+						// Maybe we should retrieve collection image in a different way.
+						imageStr := fmt.Sprintf(imagesUrlTemplate, *firstUnit)
```

Comment:
client cant to do it?

Reply:
Mobile client is "dummy", he expect to receive everything prepared for him. That is how we send image on other mobile handlers.